File: ruby/lib/steam/packets/a2a_rules_response_packet.rb
# This code is free software; you can redistribute it and/or modify it under the
# terms of the new BSD License.

autoload "SteamPacket", "steam/packets/steam_packet"
require "enumerator"

# The A2A_RULES_ResponsePacket class represents the response to a A2A_RULES
# request send to the server.
class A2A_RULES_ResponsePacket < SteamPacket
  
  # Creates a A2A_RULES response object based on the data received.
  def initialize(content_data)
    if content_data == nil
      raise Exception.new("Wrong formatted A2A_RULES response packet.")
    end
    
    super SteamPacket::A2A_RULES_RESPONSE_HEADER, content_data
    
    @rules_hash = Hash.new @content_data.get_short

    while @content_data.remaining > 0
      @rules_hash[@content_data.get_string] = @content_data.get_string
    end
  end

  # Returns the hash containing the server rules
  def get_rules_hash
    return @rules_hash
  end
end

File: php/lib/steam/packets/A2A_RULES_ResponsePacket.php
<?php
/**
 * @author Sebastian Staudt
 * @license http://www.opensource.org/licenses/bsd-license.php Modified BSD License
 * @package Steam Interface Package (PHP)
 * @subpackage SteamPacket
 * @version $Id$
 */

class A2A_RULES_ResponsePacket extends SteamPacket
{	
	/**
	 * @var String[]
	 */
	private $rulesArray;

	/**
	 * @param String $contentData
	 */
	public function __construct($contentData)
	{
		if(empty($contentData))
		{
			throw new Exception("Wrong formatted A2A_RULES response packet.");
		}
		parent::__construct(SteamPacket::A2A_RULES_RESPONSE_HEADER);
		
		$contentData = unpack("vrulesNumber/a*rulesData", $contentData);
		
		$tmpRulesArray = (explode("\0", $contentData["rulesData"]));
		for($x = 0; $x < sizeof($tmpRulesArray); $x++)
		{
			$this->rulesArray[$tmpRulesArray[$x]] = $tmpRulesArray[++$x];
		}
	}
	
	/**
	 * @return String[]
	 */
	public function getRulesArray()
	{
		return $this->rulesArray;
	}
}
?>